aten/src/ATen/core/library.cpp

```diff
@@ -54,7 +54,7 @@ CppFunction::~CppFunction() = default;
 Library::Library(Kind kind, std::string ns, c10::optional<c10::DispatchKey> k, const char* file, uint32_t line)
   : kind_(kind)
   , ns_(ns == "_" ? c10::nullopt : c10::make_optional(std::move(ns)))
-  , dispatch_key_((!k.has_value() || *k == c10::DispatchKey::CatchAll) ? c10::nullopt : k)
+  , dispatch_key_((!k.has_value() || *k == c10::DispatchKey::CatchAll) ? c10::optional<c10::DispatchKey>() : k)
```

Comment:
Considering where the crash happens, this change looks suspicious, though, to be frank,  I'm not sure what's the difference between `std::optional<>()` and `std::optional<>(std::nullopt)` constructors.

Reply:
Why not use redefining c10::nullopt to std::nullopt?

Reply:
Anyway, ignore that, the crash is due to some ABI issues between clang and gcc

Reply:
> Why not redefine c10::nullopt to std::nullopt?
already done